userservice/src/main/java/com/infodation/userservice/services/UserServiceImpl.java

```diff
@@ -6,26 +6,34 @@
 import com.infodation.userservice.repositories.UserRepository;
 import com.infodation.userservice.services.iservice.IUserService;
 import org.springframework.data.domain.Page;
-import org.springframework.data.domain.PageRequest;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
 import java.time.LocalDateTime;
-import java.util.List;
 
 @Service
 public class UserServiceImpl implements IUserService {
 
     private final UserRepository userRepository;
 
-    public UserServiceImpl(UserRepository userRepository) {
-        this.userRepository = userRepository;
-    }
     @Override
-    public Page<User> getAll(Pageable pageable) {
+    public Page<User> getAll(Pageable pageable, String name) {
+        if (name == null) {
+            name = "";
+        }
+        if (!name.isEmpty()) {
+            return userRepository.findByName(name, pageable);
+        }
         return userRepository.findAll(pageable);
```

Comment:
Can you check if we pass `name` emptyl value, UserRepository#findByName still works?
If so, you can shorten the method like:
```suggestion
       String query = Optional.ofNullable(name).orElse("");
       return userRepository.findByName(name, pageable);
```

Reply:
In line 22, I replaced 'name' with 'query', and it produced the correct result.Thank you for your advice, the code has become more concise
![image](https://github.com/user-attachments/assets/cadb2cf3-65ea-4d63-b6ed-ddac3b9959fd)
![image](https://github.com/user-attachments/assets/faf49e9c-3c63-4540-9d29-9f6123fa4811)